backend/src/nodes/nodes/image/video_frame_iterator.py

```diff
@@ -251,3 +256,28 @@ def before(index: int):
             )
 
         await context.run_while(frame_count, before, fail_fast=True)
+
+        ffmpeg_reader.stdout.close()
+        ffmpeg_reader.wait()
+        if writer["out"] is not None:
+            writer["out"].stdin.close()
+            writer["out"].wait()
+
+        if writer["copy_audio"]:
```

Comment:
I just noticed. Why is `writer` an untyped dict and not something like a dataclass instance?

Reply:
It exists only to transfer data between the iterator and the helper. It's kinda shitty but it works

Reply:
We should fix that after this PR.